client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/HeaderDecoder.java

```diff
@@ -91,6 +144,42 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
                operation.acceptResponse(in, status, this);
                checkpoint(State.READ_MESSAGE_ID);
                break;
+            case READ_CACHE_EVENT:
+               if (trace) {
+                  log.tracef("Decoding cache event %s", HotRodConstants.Names.of(receivedOpCode));
+               }
+               AbstractClientEvent cacheEvent;
+               try {
+                  cacheEvent = codec.readCacheEvent(in, channelFactory.getMarshaller(),
+                        receivedOpCode, configuration.serialWhitelist(), ctx.channel().remoteAddress());
+               } catch (Signal signal) {
+                  throw signal;
+               } catch (Throwable t) {
+                  log.unableToReadEventFromServer(t, ctx.channel().remoteAddress());
+                  throw t;
+               }
+               if (operation != null) {
+                  ((AddClientListenerOperation) operation).postponeTimeout(ctx.channel());
+               }
+               invokeEvent(cacheEvent.getListenerId(), cacheEvent);
+               checkpoint(State.READ_MESSAGE_ID);
+               break;
+            case READ_COUNTER_EVENT:
+               if (trace) {
+                  log.tracef("Decoding counter event %s", HotRodConstants.Names.of(receivedOpCode));
+               }
+               HotRodCounterEvent counterEvent;
+               try {
+                  counterEvent = codec.readCounterEvent(in);
+               } catch (Signal signal) {
+                  throw signal;
```

Comment:
Seems redundant as the outer catch does this already.

Reply:
This is not redundant, we don't want to log anything for `Signal` in the `catch (Throwable)` below